### ISSUE TITLE
Cookie prefix based on cookie name

### ISSUE THREAD

**Issue by klopfdreh** [mitre/HTTP-Proxy-Servlet#91]
Hi,

it would be very cool to be able to prefix the cookie based on its original name. The method signature needs to be changed for that:

```java
protected String getCookieNamePrefix(String originalName) {
....
}
```

For the places the renaming should be done I would suggest to put in the actual original name of the cookie.

The default implementation could be as it is, now.

WDYT?

Thanks in advance!

**Comment by dsmiley** [comment]
Sounds good to me. Submit a pull request. (don't worry about back-compatibility)

**Comment by klopfdreh** [comment]
Hope this is ok - I also checked the unit tests - all fine.

Maybe it would also be great to use the "Java Conventions" formatter of eclipse to format the source code.

**Comment by klopfdreh** [comment]
The implementation would look like:

```java
protected String getCookieNamePrefix(String name) {
   if(name.contains("lorem")){
      return "!Proxy!" + getServletConfig().getServletName();
   } else {
      return "";
   } 
}
```

**Comment by dsmiley** [comment]
I merged #92 with some changes.  At least the canonical build -- "mvn compile" is happy with respect to warnings.  

RE serialversionUid, I'm opinionated on that.  I don't add it unless I'm deliberately trying to be serializable and declare to the world that the class in question will maintain serializability and modify the constant as the class is modified in the future (e.g. new fields).  But I wish to not do that for any servlet I've written.  I think the notion of serializing servlets was a design mistake.
